std/algorithm/sorting.d

```diff
@@ -2294,9 +2294,15 @@ void topN(alias less = "a < b",
     static assert(ss == SwapStrategy.unstable,
             "Stable topN not yet implemented");
     auto heap = BinaryHeap!Range1(r1);
+    auto shadow = r2;
+    size_t i = 0;
     for (; !r2.empty; r2.popFront())
```

Comment:
No, sorry.  I was going to push it together with the unittests, but I haven't got those right yet.  In hindsight, I probably shouldn't have commented until I had pushed.